src/Build.UnitTests/FileUtilitiesRegex_Tests.cs

```diff
@@ -525,64 +525,46 @@ public void MatchLengthStartWithUncPatternNoShare()
         [Fact]
         public void UncPatternEmptyString_LegacyRegex()
         {
-            string winDirectory = string.Format("", _directoryStart);
-            string unixDirectory = string.Format("", _altDirectoryStart);
-
-            UncPattern.IsMatch(winDirectory).ShouldBe(false);
-            UncPattern.IsMatch(unixDirectory).ShouldBe(false);
+            UncPattern.IsMatch(string.Empty).ShouldBe(false);
```

Comment:
Looking at all these tests, they don't really feel substantively different from each other, and many check the same thing twice. Maybe condense into one test?

Reply:
How are those changes?

Reply:
@Forgind I disagree pretty strongly with this recommendation, because it violates the test principle that a test should fail for only one reason. Here the new test could fail if one of the implementations fails but it wouldn't be super clear from the failure which one.

@elachlan don't bother reverting though; this isn't a huge deal and you already did the unification.

Reply:
That's fair. I was borderline on suggesting just removing the legacy tests entirely—they test Regex, which isn't even in MSBuild. FileUtilitiesRegex.IsUncPattern and FileUtilitiesRegex.StartsWithUncPattern both just call FileUtilitiesRegex.StartsWithUncPatternMatchLength, so it all felt like one test to me. It's possible that won't be true in the future, though.

Reply:
Up to the team, I don't mind either way. Just let me know what you want me to do. If there isn't any changes then is this okay to merge?